### ISSUE TITLE
[Suggestion] NameChange from a list of random names between rounds

### ISSUE THREAD

**Issue by Retarded-Skid** [nullworks/cathook#166]
So instead of name stealing during the round, this will change your name to something random *(Read from a list works fine, really would be cool to steal names from users who recently disconnected)* between the rounds. This way people who don't recognize you by avatar won't follow whos who beteen game resets.

**Comment by nullifiedcat** [comment]
So, storing disconnected players' names and use them as namesteal?

**Comment by Retarded-Skid** [comment]
Pretty much. But the key part is that it switches between server restarts so nobody sees the message in chat.

**Comment by SylveonBottle** [comment]
Might be possible to prioritize names to steal based on enemy/team, early/late leaving, etc. Depending on how you look at it, switching to a person's name of one who left super early or super late into the end round results may achieve better "unnoticability".

To go even further, it might be wise to clear the list every round so that you don't get somebody who left half an hour ago. That's not to say that it wouldn't be wise to that, though - it may even be less suspicious than someone leaving the previous round immediately re-joining. In fact, one may even be able to notice the avatar change.

tl;dr Options to clear name list X rounds/minutes, setting to change name based on earliest to leave (someone you saw leave an hour ago) or someone who just left (in that case, only store 1-2 names at a time)

**Comment by BenCat07** [comment]
Adding. Will make it store recently disconnected users and change name shortly before map change, If name list > 2: pop entry

**Comment by Retarded-Skid** [comment]
:ok_hand:
